components/script/dom/window.rs

```diff
@@ -2001,7 +1998,8 @@ impl Window {
         }
 
         let document = self.Document();
-        let font_face_set = document.Fonts(can_gc);
+        document.update_fonts_state();
+        let font_face_set = document.Fonts();
```

Comment:
The goal of this work is to avoid calling Fonts in this method, since it can trigger a GC when initializing the FobtFaceSet if it dies not already exist.